languages/tree-sitter-query.scm

```diff
@@ -0,0 +1,94 @@
+; Configuration
+(#language! tree-sitter-query)
+
+; Sometimes we want to indicate that certain parts of our source text should
+; not be formatted, but taken as is. We use the leaf capture name to inform the
+; tool of this.
+[
+  (capture)
+  (identifier)
+  (string)
+] @leaf
+
+(comment) @prepend_hardline @append_hardline
```

Comment:
What about comments like this:

```scm
(foo
  (_) ; last named node
  .
)
```

I think this will probably work:

```suggestion
(comment) @prepend_input_softline @append_hardline
```

Reply:
I would rather avoid the `input_softline` predicates, since I think Topiary should be strongly opinionated. Although you're right, the comment you give as an example won't be processed correctly.
I guess until we rework the comment processing altogether, `@prepend_input_softline` will do.